recipes/unidist/meta.yaml

```
@@ -0,0 +1,131 @@
+{% set name = "unidist" %}
+{% set version = "0.1.0" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 5b2cc602604f75ed6a9323cd4cd0585a1c806d3465f50ecb7ff31772b80f2edb
+
+build:
+  # noarch disabled because the recipe now has selector for Ray, see below
+  # noarch: python
+  number: 0
+  skip: true  # [py<37]
+
+# the outputs map the unidist options on PyPI (see upstream definition
+# https://github.com/modin-project/unidist/blob/0.1.0/setup.py#L39-L55)
+# from "unidist[<option>]" to "unidist-<option>" as a conda-forge package
+outputs:
+  - name: unidist-all
+    requirements:
+      host:
+        - python
+      run:
+        - python
+        - {{ pin_subpackage('unidist', exact=True) }}
+        - {{ pin_subpackage('unidist-dask', exact=True) }}
+        # ray does not have OSX builds yet
+        - {{ pin_subpackage('unidist-ray', exact=True) }}   # [not osx]
+        - {{ pin_subpackage('unidist-mpi', exact=True) }}
+    test:
+      imports:
+        # dummy test; actual tests are in subpackages
+        - unidist
+
+  - name: unidist
+    build:
+      script: python -m pip install . --ignore-installed -vv
+    requirements:
+      host:
+        - pip
+        - python
+      run:
+        - python
+        - packaging
+        - cloudpickle
+    test:
+      imports:
+        - unidist
+      commands:
+        # unidist has multiprocessing and python backends by default so we test both
+        - python -c "import unidist; import unidist.config as cfg; cfg.Backend.put('python'); unidist.init(); ref = unidist.put(1); unidist.get(ref)"
+        - python -c "import unidist; import unidist.config as cfg; cfg.Backend.put('multiprocessing'); unidist.init(); ref = unidist.put(1); unidist.get(ref)"
+
+  - name: unidist-dask
+    build:
+      noarch: python
+    requirements:
+      host:
+        - python
+      run:
+        - python
+        - {{ pin_subpackage('unidist', exact=True) }}
+        - dask >=2.22.0
+        - distributed >=2.22.0
+    test:
+      imports:
+        - unidist
+      commands:
+        - python -c "import unidist; import unidist.config as cfg; cfg.Backend.put('dask'); unidist.init(); ref = unidist.put(1); unidist.get(ref)"
+
+  - name: unidist-ray
+    build:
+      # There is no Ray in conda-forge for MacOS yet, see conda-forge/ray-packages-feedstock#2.
+      # There is no Ray for Python >3.9 for now.
+      # TODO: when Ray support for 3.10 is released, this restriction should be lifted.
+      skip: true  # [osx or py>39]
```

This should also be `noarch: python`, the solver will take care of this.
```suggestion
    build:
      noarch: python
```

---

Wouldn't an error be when installing unidist-ray if [osx or py>39] was present?

---

No, the solver will inform the end-user, that the package cannot be installed, until the requirement is available.

---

I.e. if [osx or py>39] was present, a user would get default unidist (with multiprocessing and python backends) when installing unidist-ray on osx?

---

No, the package would not exist for that combination. They would only get the error message that this package cannot be found.

With the noarch setup, they get the error message that the package is available but a dependency isn't.

---

Got it, thanks!